### PR TITLE
Tech writer refactor of GCP deployment docs

### PR DESCRIPTION
**Pull request by sarahmaddox**
fixes https://github.com/kubeflow/website/issues/675

This PR reworks most of the GCP deployment docs, applying tech writing style and standards.

Notes:

* I have NOT fixed up the IAP / basic auth issues described in https://github.com/kubeflow/website/issues/673
* I have also NOT fixed the missing information about the option to create permanent storage https://github.com/kubeflow/website/issues/674
* I've removed the **Next Steps** doc that's currently at https://www.kubeflow.org/docs/gke/deploy/next-steps/, as it duplicates the info already given at the end of the deployment guides. I haven't added a redirect, as this doc is new and I'm guessing there are very few external references to it. I've checked for internal references from with the Kubeflow website - there are none.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/website/676)

### CODE REVIEW

**Comment by sarahmaddox** [comment]
I've also taken care to include a mention of GCP at the top of each doc, for people who may wander in off the internet and not notice the breadcrumb, therefore not realizing that these docs are GCP specific.

**Comment by sarahmaddox** [comment]
Previews:

* https://deploy-preview-676--competent-brattain-de2d6d.netlify.com/docs/started/getting-started-gke/
* https://deploy-preview-676--competent-brattain-de2d6d.netlify.com/docs/gke/deploy/
* https://deploy-preview-676--competent-brattain-de2d6d.netlify.com/docs/gke/deploy/project-setup/
* https://deploy-preview-676--competent-brattain-de2d6d.netlify.com/docs/gke/deploy/oauth-setup/
* https://deploy-preview-676--competent-brattain-de2d6d.netlify.com/docs/gke/deploy/deploy-ui/
* https://deploy-preview-676--competent-brattain-de2d6d.netlify.com/docs/gke/deploy/deploy-cli/
* https://deploy-preview-676--competent-brattain-de2d6d.netlify.com/docs/gke/deploy/delete-cli/
* https://deploy-preview-676--competent-brattain-de2d6d.netlify.com/docs/gke/deploy/delete-ui/
https://deploy-preview-676--competent-brattain-de2d6d.netlify.com/docs/gke/deploy/reasons/

**Comment by sarahmaddox** [comment]
This PR is now ready for review. I'm applying a hold to allow for multiple reviewers.

/hold

File: content/docs/gke/deploy/reasons.md [hunk 1]
```diff
@@ -1,29 +1,39 @@
...
     large resource requirements.
   * [Cloud Identity-Aware Proxy (Cloud IAP)](https://cloud.google.com/iap/) 
     makes it easy to securely connect to Jupyter and other
     web apps running as part of Kubeflow.
-  * Basic auth service supports simple username/password access to your 
-    Kubeflow. It is an alternative to Cloud IAP service:
-    * We recommend IAP for production and enterprise workloads.
-    * Consider basic auth only when trying to test out Kubeflow and use it 
+  * Kubeflow's basic authentication service supports simple username/password
+    access to your Kubeflow resources. Basic auth is an alternative to Cloud
```
Comment (aikidave) [inline content/docs/gke/deploy/reasons.md]: Should this reference to 'basic auth' change to "basic authentication"?

Comment (sarahmaddox) [reply content/docs/gke/deploy/reasons.md]: I decided to leave "basic auth" in subsequent mentions, after using the full term once in this section, given that "basic auth" is a fairly well known term. Happy to change it if you think it'd be clearer, perhaps in a subsequent PR?

Comment (aikidave) [reply content/docs/gke/deploy/reasons.md]: I defer to your judgment! I just wanted to make sure that this was intentional.

Comment (sarahmaddox) [reply content/docs/gke/deploy/reasons.md]: Thanks :) I'll leave it as it is for now.

**Comment by abhi-g** [comment]
Thanks Sarah.

/lgtm
/approve

**Pull request merged**